### ISSUE TITLE
Unexpected error with non-existing successfull builds

### ISSUE THREAD

**Issue by DavidPLamas** [nrwl/nx-orb#6]
## Orb version:
1.1.2

## What happened:
I set up the circleci configuration file for the first time using the guidelines in the [documentation ](https://circleci.com/developer/orbs/orb/nrwl/nx) but it is not working as expected. It looks like the step nx/set-shas has an error and the rest of the workflow does not proceed the way it was supposed to be.

## Expected behavior:
The build should finish with success

## Additional Information:
Here is my circleci configuration file:

```
version: 2.1

# Orbs are reusable packages of CircleCI configuration that you may share across projects, enabling you to create encapsulated, parameterized commands, jobs, and executors that can be used across multiple projects.
# See: https://circleci.com/docs/2.0/orb-intro/
orbs:
  nx: nrwl/nx@1.1.2

jobs:
  build:
    docker:
      - image: cimg/node:14.17-browsers
    steps:
      # ...CI steps like checkout, install...
      - checkout
      - run:
          command: yarn install --frozen-lockfile
          name: Install dependencies
      - nx/set-shas
      - run:
          command: yarn nx affected --target=build --base=$NX_BASE --parallel --max-parallel=3
          name: Run Builds
      - run:
          command: yarn nx affected --target=test --base=$NX_BASE
          name: Run Unit Tests

# Invoke jobs via workflows
# See: https://circleci.com/docs/2.0/configuration-reference/#workflows
workflows: 
  version: 2
  build:
    jobs:
      - build
```

Here is the output from circleci:
![image](https://user-images.githubusercontent.com/8373746/140433620-dbc79c27-5098-4461-aeba-dddca204e72b.png)

**Comment by DavidPLamas** [comment]
The problem was fixed by introducing the CIRCLE_API_TOKEN environment variable, which is in fact documented in the Readme file of this repository. You should probably update [CircleCI official documentation](https://circleci.com/developer/orbs/orb/nrwl/nx) page with this feature as well.

**Comment by cdierkens** [comment]
@DavidPLamas I have CIRCLE_API_TOKEN set in my project environment variables but I'm still getting the error. Is there something I need to do to make CIRCLE_API_TOKEN available to the orb?

**Comment by DavidPLamas** [comment]
> @DavidPLamas I have `CIRCLE_API_TOKEN` set in my project environment variables but I'm still getting the error. Is there something I need to do to make `CIRCLE_API_TOKEN` available to the orb?

Yes, there is. Go to your project settings in Circle CI and add that new environmental variable.

**Comment by cdierkens** [comment]
>  I have `CIRCLE_API_TOKEN` set in my project environment variables but I'm still getting the error.

Something else is afoot. I already have that set.

**Comment by DavidPLamas** [comment]
> > I have `CIRCLE_API_TOKEN` set in my project environment variables but I'm still getting the error.
> 
> Something else is afoot. I already have that set.

I haven't worked on that project for about a month, so something might have changed meanwhile... 
This is my current pipeline:

```
version: 2.1

# Orbs are reusable packages of CircleCI configuration that you may share across projects, enabling you to create encapsulated, parameterized commands, jobs, and executors that can be used across multiple projects.
# See: https://circleci.com/docs/2.0/orb-intro/
orbs:
  nx: nrwl/nx@1.1.2

jobs:
  build:
    docker:
      - image: cimg/node:14.17-browsers
    steps:
      # ...CI steps like checkout, install...
      - checkout
      - run:
          command: yarn install --frozen-lockfile
          name: Install dependencies
      - nx/set-shas
      - run:
          command: yarn affected:build --base=$NX_BASE --parallel --max-parallel=3
          name: Run Builds
      # - run:
      #     command: yarn nx affected --target=test --base=$NX_BASE
      #     name: Run Unit Tests

# Invoke jobs via workflows
# See: https://circleci.com/docs/2.0/configuration-reference/#workflows
workflows: 
  version: 2
  build:
    jobs:
      - build
```

**Comment by cdierkens** [comment]
@DavidPLamas Thanks for sharing the code? If you still have access, can you check what permissions you gave to your API key?

Here is some information from my end just for brevity...

Set in env vars:

![image](https://user-images.githubusercontent.com/4812409/146247542-ccefce3e-bfa1-4def-be30-9befabe5a0b0.png)

Set when I ssh into the job:

```
circleci@ff360e330413:~$ echo $CIRCLE_API_TOKEN
************************************0da2
```

And matches what I got back from the Project settings

![image](https://user-images.githubusercontent.com/4812409/146247688-a22dbe2e-3308-4dbf-9576-90bbed5ec2c4.png)


Note that I was getting false positives by not passing `NX_HEAD` directly. 

![image](https://user-images.githubusercontent.com/4812409/146248536-bba787e2-0d4e-4a8c-81f3-19989f8f0c7e.png)

**Comment by DavidPLamas** [comment]
@cdierkens The API Token I'm using is a personal token and not a project specific token. That might be the difference between you and me... 
Go to your profile, create a "Personal API Token" and then try using that token instead of the project specific one. 

I'm not sure if this is the correct approach but it works :)

**Comment by rdraward-mindbridge** [comment]
@cdierkens @DavidPLamas I just finished dealing with the same issue where a project level token was not working. After ssh'ing into the build box to hit the Circle API directly, and investigating the error response a bit more I was able to dig up [this Circle API support page](https://support.circleci.com/hc/en-us/articles/360060360811-CircleCI-API-v2-returns-404-Not-Found-) that states that project tokens are not supported on v2 of the Circle API that is used by the orb.

Setting up a machine user and creating a personal token works like a charm!

@meeroslav Perhaps a small change to the README to specify that a personal Circle token is needed would be helpful.
